tmt/base.py

```diff
@@ -1722,6 +1735,20 @@ def _initialize_data_directory(self) -> None:
             f"Create the data directory '{self.data_directory}'.", level=2)
         self.data_directory.mkdir(exist_ok=True, parents=True)
 
+    def _initialize_env_file(self) -> None:
+        """
+        Create the plan environment file
+
+        Environment variables stored in this file are sourced
+        before the execute step and are available to all
+        subsequent steps.
+        """
+        assert self.data_directory is not None  # narrow type
+        self.env_file = self.data_directory / "variables.env"
+        self.debug(
+            f"Create the environment file '{self.env_file}'.", level=2)
+        self.env_file.touch(exist_ok=True)
```

Comment:
This would be a nice cached property - the first time it's accessed, the file would be created.

```python
@tmt.utils.cached_property
def env_file(self) -> Path:
    assert self.data_directory is not None  # narrow type

    env_file_path self.data_directory / "variables.env"
    env_file_path.touch(exist_ok=True)

    self.debug(f"Create the environment file '{env_file_path}'.", level=2)

    return env_file_path
```

Path doesn't change, it's a property, no need to call it explicitly when we can forget about it, and it'll materialize the first time tmt tries to add it to a step's environment.

one think I'm worried about though is the name - there's already `environment` and `environment-file`, not `env_file`. It would be nice to tell the difference between `environment-file` and `env_file`, but I have no idea what to propose :/

Reply:
Thank you for the suggestion. I added the cached property.

I agree that the name is quite confusing, but I'm also not sure what would be the best name for it.